lms/static/js/learner_dashboard/views/entitlement_unenroll_view.js

```diff
@@ -0,0 +1,159 @@
+                    });
+                },
+
+                resetModal: function() {
+                    this.$submitButton.removeData();
+                    this.$submitButton.prop('disabled', false);
+                    this.$trackInfo.empty();
+                    this.$refundInfo.empty();
+                    this.$errorInfo.css('display', 'none');
+                    this.$errorInfo.empty();
+                },
+
+                setError: function(message) {
+                    this.$submitButton.prop('disabled', true);
+                    this.$errorInfo.empty();
+                    HtmlUtils.setHtml(
+                        this.$errorInfo,
+                        message
+                    )
+                    this.$errorInfo.css('display', 'block');
+                },
+
+                setTrackInfo: function(courseName, courseNumber) {
+                    this.$trackInfo.empty();
+                    HtmlUtils.setHtml(
+                        this.$trackInfo,
+                        HtmlUtils.interpolateHtml(
+                            gettext('Are you sure you want to unenroll from {courseName} ({courseNumber})?'),
+                            {
+                                courseName: HtmlUtils.joinHtml(
+                                    HtmlUtils.HTML('<span id="entitlement-unenrollment-course-name">'),
+                                    courseName,
+                                    HtmlUtils.HTML('</span>')
+                                ),
+                                courseNumber: HtmlUtils.joinHtml(
+                                    HtmlUtils.HTML('<span id="entitlement-unenrollment-course-number">'),
+                                    courseNumber,
+                                    HtmlUtils.HTML('</span>')
+                                )
+                            }
+                        )
+                    );
+                },
+
+                setRefundInfo: function(isRefundable) {
+                    var message = null;
+
+                    if (isRefundable) {
+                        message = gettext('You will be refunded the amount you paid.');
+                    } else {
+                        message = gettext('You will not be refunded the amount you paid.');
+                    }
+
+                    this.$refundInfo.empty();
+                    HtmlUtils.setHtml(this.$refundInfo, message);
+                },
+
+                setSubmitData: function(apiEndpoint, isRefundable) {
+                    this.$submitButton.removeData();
+                    this.$submitButton.data('entitlementApiEndpoint', apiEndpoint);
+                    this.$submitButton.data('entitlementIsRefundable', isRefundable);
+                },
+
+                onComplete: function(xhr, textStatus) {
+                    if (xhr.status === 204) {
+                        window.location.href = this.dashboardUrl
+                    } else {
```

Comment:
Note: I plan to change this to handle error states more appropriately before this merges. (Ex: redirect to sign-in on authentication error).